src/auth/auth.controller.ts

```diff
@@ -0,0 +1,60 @@
+import { instanceToPlain, plainToInstance } from 'class-transformer';
+import { validate } from 'class-validator';
+import { Request, Response } from 'express';
+
+import { Controller } from '../common';
+
+import { AuthService } from './auth.service';
+import { LoginDto, RegisterDto } from './dto';
+
+export class AuthController extends Controller {
+    constructor() {
+        super('/auth');
+
+        const router = this.getRouter();
+
+        router.post('/login', this.login);
+        router.post('/register', this.register);
+        router.post('/logout', this.logout);
+    }
+
+    async login(req: Request, res: Response) {
+        const body = plainToInstance(LoginDto, req.body as LoginDto);
+        const errors = await validate(body);
+        if (errors.length > 0) {
+            return res.status(400).json(errors);
```

Comment:
we could use enum for status codes, it would be more readable to operate on enums like "NotFound" etc instead of 401,404,400...